addons/survey/tests/test_survey_invite.py

```diff
@@ -194,3 +194,30 @@ def test_survey_invite_token_internal(self):
             set(answers.mapped('email')),
             set([self.user_emp.email]))
         self.assertEqual(answers.mapped('partner_id'), self.user_emp.partner_id)
+
+    def test_survey_invite_token_by_email_nosignup(self):
+        """
+        Case: have multiples partners with the same email address
+        If I set one email address, I expect one email to be sent
+        """
+
+        first_partner = self.env['res.partner'].create({
+            'name': 'Test 1',
+            'email': 'test@example.com',
+        })
+
+        self.env['res.partner'].create({
+            'name': 'Test 2',
+            'email': 'TEST@example.COM',
```

Comment:
Just to be annoying, did you try with an email like
```
"Raoul Poilvache" <TEST@example.COM>
```
? Indeed people may put formatted emails in an email field, not just a pure email.

Reply:
Oh right, I didn't know we could do that